alicloud/data_source_alicloud_snapshots.go

```diff
@@ -0,0 +1,280 @@
+package alicloud
+
+import (
+	"fmt"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+	"regexp"
+)
+
+func dataSourceAlicloudSnapshots() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudSnapshotsRead,
+		Schema: map[string]*schema.Schema{
+			"instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"disk_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"encrypted": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"ids": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				MinItems: 1,
+				MaxItems: 100,
+			},
+			"name_regex": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				//TODO 字符串换成常量
+				ValidateFunc: validateAllowedStringValue([]string{"progressing", "accomplished", "failed", "all"}),
+				Default:      "all",
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				//TODO 字符串换成常量
+				ValidateFunc: validateAllowedStringValue([]string{"auto", "user", "all"}),
+				Default:      "all",
+			},
+			"source_disk_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				//TODO 字符串换成常量
+				ValidateFunc: validateAllowedStringValue([]string{"System", "Data"}),
+				Default:      "", //TODO 确定默认值
+			},
+			"usage": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				//TODO 字符串换成常量
+				ValidateFunc: validateAllowedStringValue([]string{"image", "disk", "image_disk", "node"}),
+				Default:      "", //TODO 确定默认值
+			},
+			"output_file": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": tagsSchema(),
+			"snapshots": &schema.Schema{
```

Comment:
Please add an extra output parameter `names` that is used to store the snapshots' name.